apps/docs/src/app/platform/component-docs/platform-table/platform-table-examples/platform-table-tree-example.component.ts

```diff
@@ -0,0 +1,339 @@
+import { Component } from '@angular/core';
+import { Observable, of } from 'rxjs';
+
+import { FdDate } from '@fundamental-ngx/core';
+import {
+    TableDataSource,
+    TableDataProvider,
+    TableState,
+    TableRowSelectionChangeEvent
+} from '@fundamental-ngx/platform';
+
+@Component({
+    selector: 'fdp-platform-table-tree-example',
+    templateUrl: './platform-table-tree-example.component.html'
+})
+export class PlatformTableTreeExampleComponent {
+    source: TableDataSource<ExampleItem>;
+
+    constructor() {
+        this.source = new TableDataSource(new TableDataProviderExample());
+    }
+
+    alert(message: string): void {
+        alert(message);
+    }
+
+    onRowSelectionChange(event: TableRowSelectionChangeEvent<ExampleItem>): void {
+        console.log(event);
+    }
+}
+
+interface BaseItem {
+    name: string;
+}
+
+export interface ExampleItem extends BaseItem {
+    name: string;
```

Comment:
You do not need `name` in the `ExampleItem` and `ExampleItem` at the same time, or you do not need `BaseItem` at all